Doesn't support install with yarn

`package.json`
```json
{
  "scripts": {
    "preinstall": "run-script-os",
    "install:win32": "echo 'win32'",
    "install:darwin:linux": "echo 'darwin/linux'"
  }
}
```

Running: `yarn` (on its own) is equivalent to running `yarn install`, but fails with the following error:
```bash
yarn install v1.13.0
$ run-script-os
npm ERR! missing script: undefined:darwin
```

While: `yarn install` works correctly:
```bash
yarn install v1.13.0
$ run-script-os

> atlas-web-and-tv@1.0.0 install:darwin:linux /Users/SFL16/Development/atlas-web-and-tv
> echo 'darwin/linux'

darwin/linux
[1/5] 🔍  Validating package.json...
...
```

--- a/README.md
+++ b/README.md
@@ -46,6 +46,16 @@ del whatever you want in Windows 32/64
 You can combine OS tags and rm all the things!
 ```
 
+### Override detection settings for linux-based shells on Windows
+
+By default, run-script-os will detect cygwin/git bash as Linux. If you would rather your platform be detected as Windows under these environments:
+
+Set environment variable:
+
+```
+RUN_OS_WINBASH=true
+```
+
 ### NPM Scripts Order
 When you call a script like `npm test`, npm will first call `pretest` if it exists. It will then call `test`, which, if you are using `run-script-os`, it will then call `npm run test:YOUR OS`, which in turn will call `pretest:YOUR OS` before actually running `test:YOUR OS`. Then `posttest:YOUR OS` will run, and then after that `posttest` will finally execute.
 

--- a/README.md
+++ b/README.md
@@ -16,6 +16,7 @@ Then create OS specific scripts. In the example below, you can see:
 
 * `test:win32`
 * `test:linux:darwin`
+* `test:default`
 
 Those can have OS specific logic.
 
@@ -26,8 +27,9 @@ Those can have OS specific logic.
   "scripts": {
     ...
     "test": "run-script-os",
-    "test:win32": "echo 'del whatever you want in Windows 32/64'", 
+    "test:win32": "echo 'del whatever you want in Windows 32/64'",
     "test:darwin:linux": "echo 'You can combine OS tags and rm all the things!'",
+    "test:default": "echo 'This will run on any platform that does not have its own script'"
     ...
   },
   ...
@@ -45,6 +47,15 @@ del whatever you want in Windows 32/64
 > npm test
 You can combine OS tags and rm all the things!
 ```
+
+### Aliases
+
+You can use the following aliases:
+
+* `:windows` - Alias for win32
+* `:macos` - Alias for darwin
+* `:nix` - This will run on anything considered to be a *nix OS (aix, darwin, freebsd, linux, openbsd, sunos, android)
+* `:default` - This will run if no platform-specific scripts are found
 
 ### Override detection settings for linux-based shells on Windows
 

--- a/README.md
+++ b/README.md
@@ -53,7 +53,7 @@ By default, run-script-os will detect cygwin/git bash as Windows. If you would r
 Set environment variable:
 
 ```
-RUN_OS_WINBASH=true
+RUN_OS_WINBASH_IS_LINUX=true
 ```
 
 ### NPM Scripts Order

--- a/README.md
+++ b/README.md
@@ -48,7 +48,7 @@ You can combine OS tags and rm all the things!
 
 ### Override detection settings for linux-based shells on Windows
 
-By default, run-script-os will detect cygwin/git bash as Linux. If you would rather your platform be detected as Windows under these environments:
+By default, run-script-os will detect cygwin/git bash as Windows. If you would rather your platform be detected as Linux under these environments:
 
 Set environment variable:
 

--- a/index.js
+++ b/index.js
@@ -32,6 +32,9 @@ if (!(options[0] === "run" || options[0] === "run-script")) {
     options.unshift("run");
 }
 
+// Check for yarn without install command; fixes #13
+if (process.env.npm_config_user_agent.includes('yarn') && !options[1]) options[1] = 'install';
+
 let osCommand = `${options[1]}:${platform}`
 if (!(osCommand in scripts)) {
     let regex = new RegExp(`^(${options[1]}):([a-zA-Z0-9-]*:)*(${platform})(:[a-zA-Z0-9-]*)*$`, "g");

--- a/index.js
+++ b/index.js
@@ -11,7 +11,7 @@ const spawn = require("child_process").spawn;
 let scripts;
 
 // Switch to linux platform if cygwin/gitbash detected (fixes #7)
-// Allow overriding this behavior (fixes #8)
+// Allow overriding this behavior (fixes #11)
 let platform = process.platform
 if (process.env.RUN_OS_WINBASH) {
   let shell = process.env.SHELL || process.env.TERM

--- a/index.js
+++ b/index.js
@@ -6,17 +6,18 @@ if (!process.env["npm_config_argv"]) {
     console.log("This is meant to be run from within npm script. See https://github.com/charlesguse/run-script-os");
     return;
 }
+
 const spawn = require("child_process").spawn;
 
 let scripts;
 
 // Switch to linux platform if cygwin/gitbash detected (fixes #7)
 // Allow overriding this behavior (fixes #11)
-let platform = process.platform
+let platform = process.platform;
 if (process.env.RUN_OS_WINBASH_IS_LINUX) {
-  let shell = process.env.SHELL || process.env.TERM
-  shell = shell && shell.match("bash.exe") ? "bash.exe" : shell
-  platform = shell && ["bash.exe", "cygwin"].includes(shell) ? "linux" : process.platform
+  let shell = process.env.SHELL || process.env.TERM;
+  shell = shell && shell.match("bash.exe") ? "bash.exe" : shell;
+  platform = shell && ["bash.exe", "cygwin"].includes(shell) ? "linux" : process.platform;
 }
 
 if (platform === "win32") {
@@ -30,9 +31,10 @@ let options = npmArgs.original;
 if (!(options[0] === "run" || options[0] === "run-script")) {
     options.unshift("run");
 }
+
 let osCommand = `${options[1]}:${platform}`
 if (!(osCommand in scripts)) {
-    let regex = new RegExp(`^(${options[1]}):([a-zA-Z0-9-]*:)*(${platform})(:[a-zA-Z0-9-]*)*$`, "g")
+    let regex = new RegExp(`^(${options[1]}):([a-zA-Z0-9-]*:)*(${platform})(:[a-zA-Z0-9-]*)*$`, "g");
     for (let command in scripts) {
         if (command.match(regex)) {
             osCommand = command;
@@ -40,6 +42,7 @@ if (!(osCommand in scripts)) {
         }
     }
 }
+
 options[1] = osCommand;
 
 let platformSpecific;

--- a/index.js
+++ b/index.js
@@ -13,7 +13,7 @@ let scripts;
 // Switch to linux platform if cygwin/gitbash detected (fixes #7)
 // Allow overriding this behavior (fixes #8)
 let platform = process.platform
-if (!process.env.RUN_OS_WINBASH) {
+if (process.env.RUN_OS_WINBASH) {
   let shell = process.env.SHELL || process.env.TERM
   shell = shell && shell.match("bash.exe") ? "bash.exe" : shell
   platform = shell && ["bash.exe", "cygwin"].includes(shell) ? "linux" : process.platform

--- a/index.js
+++ b/index.js
@@ -18,7 +18,7 @@ if (process.env.RUN_OS_WINBASH_IS_LINUX) {
   platform = shell && ["bash.exe", "cygwin"].includes(shell) ? "linux" : process.platform;
 }
 
-const scripts = require(path.join(process.env.PWD, "package.json")).scripts;
+const scripts = require(path.join(process.cwd(), "package.json")).scripts;
 
 let npmArgs = JSON.parse(process.env["npm_config_argv"]);
 let options = npmArgs.original;

--- a/index.js
+++ b/index.js
@@ -13,7 +13,7 @@ let scripts;
 // Switch to linux platform if cygwin/gitbash detected (fixes #7)
 // Allow overriding this behavior (fixes #11)
 let platform = process.platform
-if (process.env.RUN_OS_WINBASH) {
+if (process.env.RUN_OS_WINBASH_IS_LINUX) {
   let shell = process.env.SHELL || process.env.TERM
   shell = shell && shell.match("bash.exe") ? "bash.exe" : shell
   platform = shell && ["bash.exe", "cygwin"].includes(shell) ? "linux" : process.platform

--- a/index.js
+++ b/index.js
@@ -10,10 +10,14 @@ const spawn = require("child_process").spawn;
 
 let scripts;
 
-// Switch to linux platform if cygwin/gitbash detected
-let shell = process.env.SHELL || process.env.TERM
-shell = shell && shell.match("bash.exe") ? "bash.exe" : shell
-const platform = shell && ["bash.exe", "cygwin"].includes(shell) ? "linux" : process.platform
+// Switch to linux platform if cygwin/gitbash detected (fixes #7)
+// Allow overriding this behavior (fixes #8)
+let platform = process.platform
+if (!process.env.RUN_OS_WINBASH) {
+  let shell = process.env.SHELL || process.env.TERM
+  shell = shell && shell.match("bash.exe") ? "bash.exe" : shell
+  platform = shell && ["bash.exe", "cygwin"].includes(shell) ? "linux" : process.platform
+}
 
 if (platform === "win32") {
     scripts = require("./package.json").scripts;

--- a/index.js
+++ b/index.js
@@ -3,13 +3,11 @@
 const path = require("path");
 
 if (!process.env["npm_config_argv"]) {
-    console.log("This is meant to be run from within npm script. See https://github.com/charlesguse/run-script-os");
-    return;
+  console.log("This is meant to be run from within npm script. See https://github.com/charlesguse/run-script-os");
+  return;
 }
 
 const spawn = require("child_process").spawn;
-
-let scripts;
 
 // Switch to linux platform if cygwin/gitbash detected (fixes #7)
 // Allow overriding this behavior (fixes #11)
@@ -20,41 +18,65 @@ if (process.env.RUN_OS_WINBASH_IS_LINUX) {
   platform = shell && ["bash.exe", "cygwin"].includes(shell) ? "linux" : process.platform;
 }
 
-if (platform === "win32") {
-    scripts = require("./package.json").scripts;
-} else {
-    scripts = require(path.join(process.env.PWD, "package.json")).scripts;
-}
+const scripts = require(path.join(process.env.PWD, "package.json")).scripts;
 
 let npmArgs = JSON.parse(process.env["npm_config_argv"]);
 let options = npmArgs.original;
 if (!(options[0] === "run" || options[0] === "run-script")) {
-    options.unshift("run");
+  options.unshift("run");
 }
 
 // Check for yarn without install command; fixes #13
 if (process.env.npm_config_user_agent.includes('yarn') && !options[1]) options[1] = 'install';
 
-let osCommand = `${options[1]}:${platform}`
+let osCommand = `${options[1]}:${platform}`;
+let foundMatch = true;
+
 if (!(osCommand in scripts)) {
-    let regex = new RegExp(`^(${options[1]}):([a-zA-Z0-9-]*:)*(${platform})(:[a-zA-Z0-9-]*)*$`, "g");
-    for (let command in scripts) {
-        if (command.match(regex)) {
-            osCommand = command;
-            break;
-        }
+  foundMatch = false;
+  let regex = new RegExp(`^(${options[1]}):([a-zA-Z0-9-]*:)*(${platform})(:[a-zA-Z0-9-]*)*$`, "g");
+  for (let command in scripts) {
+    if (command.match(regex)) {
+      osCommand = command;
+      foundMatch = true;
+      break;
     }
+  }
+}
+
+// Check for :windows alias; fixes #3
+if (!foundMatch && (`${options[1]}:windows` in scripts) && platform === 'win32') {
+  osCommand = `${options[1]}:windows`;
+  foundMatch = true;
+}
+
+// Check for :macos alias; fixes #3
+if (!foundMatch && (`${options[1]}:macos` in scripts) && platform === 'darwin') {
+  osCommand = `${options[1]}:macos`;
+  foundMatch = true;
+}
+
+// Check for :nix script; fixes #3
+if (!foundMatch && (`${options[1]}:nix` in scripts) && ['aix', 'darwin', 'freebsd', 'linux', 'openbsd', 'sunos', 'android'].includes(platform)) {
+  osCommand = `${options[1]}:nix`;
+  foundMatch = true;
+}
+
+// Check for :default script if no exact match; fixes #3
+if (!foundMatch && (`${options[1]}:default` in scripts)) {
+  osCommand = `${options[1]}:default`;
+  foundMatch = true;
 }
 
 options[1] = osCommand;
 
 let platformSpecific;
 if (platform === "win32") {
-    platformSpecific = spawn("npm.cmd", options, { shell: true, stdio: "inherit"});
+  platformSpecific = spawn("npm.cmd", options, { shell: true, stdio: "inherit"});
 } else {
-    platformSpecific = spawn("npm", options, { shell: true, stdio: "inherit" });
+  platformSpecific = spawn("npm", options, { shell: true, stdio: "inherit" });
 }
 
 platformSpecific.on("exit", (code) => {
-    process.exit(code);
+  process.exit(code);
 });

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "run-script-os",
-  "version": "1.0.5",
+  "version": "1.0.6",
   "description": "run-script-os is a tool that will let you use generic npm script commands that will pass through to os specific commands.",
   "main": "index.js",
   "scripts": {

